### DISCUSSION TITLE
How to add more text strings for internationalization?

### DISCUSSION THREAD

**Discussion by blanquerna** [datenstrom/community#841] [Ask a question] [answered]
Datenstrom-Yellow I'm liking it a lot because it has a solid frame that's easy to customize.

I think that the internationalization of the portal is concentrated in the translations and with the function 

> $this->yellow->language->getTextHtml()

The different language schemes have 3 blog entrys:

> [ "BlogBy: by" ] - [ "BlogTag: Tags:" ] - [ "BlogMore: Read more…" ]

For the regular operation of the blog, I think it would have 4 more entries, remaining as:

> [ "BlogBy: by" ] - [ "BlogMore: Read more…" ]
> [ "BlogShare: Share on" ] - [ "BlogRecent: Recent posts" ] - [ "BlogAuthors: Authors" ] - [ "BlogTags: Tags" ]

Maybe also 

> [ "BlogYear: Year" ] - [ "BlogAuthor: Author" ] - [ "BlogTag: Tag" ]

The current entry [ "BlogTag: Tags:" ] should remove the trailing colon and be included in the layout where needed

Can you consider expanding the language entries in a future update?

**Comment by schulle4u** [comment, marked as answer]
Can you provide a real-world content or layout example in which these language entries would be useful? As for blog shortcuts I would simply use a heading or other surrounding text to indicate them. 

Anyway you can always [change the system](https://datenstrom.se/yellow/help/how-to-change-the-system) and create your own language settings in `system/extensions/yellow-language.ini`. Just use the API to include them in a layout file like any other language setting.

**Comment by blanquerna** [comment]
Thanks for reminding me of the system/extensions/yellow-language.ini file.

I have entered these inputs into it and it works perfectly with the standard function. It even overwrites other language file entries if they are duplicates. So I can perfectly use the standard functions of the system and this is good.

I think it would be interesting to complete the basic messages of a blog. When analyzing Themes with sidebar I have seen that they used these messages in the English language and to modify them I had to modify the respective layouts to my language, only to my language.

The option to use yellow-language.ini will only work for the language it deblares and not others.

I think the entries I've suggested for a future update might fix it.

**Reply by markseuffert** [reply]
Have a look at these API functions:

* `system->setDefault()` for system settings, each setting a pair with key and value
* `language->setDefaults()` for language settings, you can provide multiple languages

Here's an example code for a new extension:

```
    // Handle initialisation
    public function onLoad($yellow) {
        $this->yellow = $yellow;
        $this->yellow->system->setDefault("helloworldMessage", "Hello World");
        $this->yellow->system->setDefault("helloworldSpeed", "100");
        $this->yellow->language->setDefaults(array(
            "Language: en",
            "HelloworldBlogShare: Share on",
            "HelloworldBlogRecent: Recent blog posts",
            "Language: de",
            "HelloworldBlogShare: Teile auf",
            "HelloworldBlogRecent: Neuste Blogbeiträge",
            "Language: sv",
            "HelloworldBlogShare: Dela på",
            "HelloworldBlogRecent: Senaste blogginlägg"));
    }
```

We're making life a little easier for translators by collecting the language settings of published extensions, especially those extensions included in the [standard installation](https://github.com/datenstrom/yellow). With the idea that the [bulk of the translation is done in one place](https://github.com/annaesvensson/yellow-language). But there's more. Any extension can bring its own default language settings and the rest of the API works exactly the same way. 

Hope this helps.
